src/main/java/net/glowstone/command/minecraft/SpawnPointCommand.java

```diff
@@ -0,0 +1,119 @@
+package net.glowstone.command.minecraft;
+
+import com.google.common.collect.ImmutableList;
+import net.glowstone.command.CommandTarget;
+import net.glowstone.command.CommandUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.defaults.VanillaCommand;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SpawnPointCommand extends VanillaCommand {
+
+    public SpawnPointCommand() {
+        super("spawnpoint", "Sets the spawn point for a player.", "/spawnpoint OR /spawnpoint <player> OR /spawnpoint <player> <x> <y> <z>", Collections.emptyList());
+        setPermission("minecraft.command.spawnpoint");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (!testPermission(sender)) return false;
+
+        if (args.length != 0 && args.length != 1 && args.length < 4) {
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            return false;
+        }
+
+        final String playerPattern = args.length >= 1 ? args[0] : null;
+        List<Player> targets;
+        Location spawnLocation = null;
+
+        // Manage player(s)
+        if (playerPattern == null) { // Default player, set to current one (if a player)
+            if (sender instanceof Player) {
+                targets = ImmutableList.of((Player) sender);
+            } else {
+                sender.sendMessage(ChatColor.RED + "Current sender is not a player.");
```

Comment:
Replace this with `You must specify which player you wish to perform this action on.`